Source/Blazorise/Components/Carousel/Carousel.razor.cs

```diff
@@ -269,79 +257,71 @@ private async void OnLocalizationChanged( object sender, EventArgs eventArgs )
             await InvokeAsync( StateHasChanged );
         }
 
-        private void InitializeTimer()
+        private Task InitializeTimer( double? interval = null )
         {
-            Timer = new( Interval );
-            Timer.Elapsed += OnTimerEvent;
-            Timer.AutoReset = true;
-        }
+            Timer = new( TimeSpan.FromMilliseconds( interval ?? Interval ) );
 
-        private void InitializeTransitionTimer()
-        {
-            TransitionTimer = new( 2000 );
-            TransitionTimer.Elapsed += OnTransitionTimerEvent;
-            TransitionTimer.AutoReset = false;
+            Task.Run( async () =>
+            {
+                while ( await Timer.WaitForNextTickAsync() )
+                {
+                    if ( AutoPlayEnabled )
+                        await OnTimerEvent( this, null );
+                };
+            } );
+            return Task.CompletedTask;
         }
 
-        private void SetTimer()
+        private Task InitializeTransitionTimer()
         {
-            TimerEnabled = ( Interval > 0 );
+            TransitionTimer = new( TimeSpan.FromMilliseconds( 2000 ) );
 
-            if ( Timer == null && TimerEnabled )
+            Task.Run( async () =>
```

Comment:
Do we still need `Task.Run`?

Reply:
Well since instead of callbacks, it setups a while.loop it needs to be running on a background thread. Suggestions?

Reply:
I think just having `while ( await Timer.WaitForNextTickAsync() )` will work.

Reply:
Am I missing something? The while is a sync operation it'll lock execution of the method until completed...

Reply:
Exactly my point. the Outter method (`OnInitializedAsync`) will not return until the timer is disposed of. Now I don't know if that has negative consequences or not.

Reply:
Are you sure it will not return? I need to confirm it....

Reply:
Pretty sure... just debug it. Why would it, again while is a sync operation.